diff --git a/t/t5322-pack-objects-sparse.sh b/t/t5322-pack-objects-sparse.sh
--- a/t/t5322-pack-objects-sparse.sh
+++ b/t/t5322-pack-objects-sparse.sh
@@ -109,7 +109,7 @@ test_expect_success 'non-sparse pack-objects' '
 '

 # --sparse is enabled by default by pack.useSparse
-test_expect_success 'sparse pack-objects' '
+test_expect_failure 'sparse pack-objects' '
 	GIT_TEST_PACK_SPARSE=-1 &&
 	git rev-parse			\
 		topic1			\
@@ -123,7 +123,7 @@ test_expect_success 'sparse pack-objects' '
 	test_cmp expect_sparse_objects.txt sparse_objects.txt
 '

-test_expect_success 'pack.useSparse enables algorithm' '
+test_expect_failure 'pack.useSparse enables algorithm' '
 	git config pack.useSparse true &&
 	git pack-objects --stdout --revs <packinput.txt >sparse.pack &&
 	git index-pack -o sparse.idx sparse.pack &&
